### PR TITLE
DYN-5189 update notifications number

### PR DESCRIPTION
**Pull request by filipeotero**
### Purpose

This PR is to count the number of notifications by requesting the backend at the dynamo side, counting and sending it to the webApp.
The implementation depends on this PR: https://github.com/DynamoDS/NotificationCenter/pull/24

### Declarations

Check these if you believe they are true

- [X] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### Reviewers

@QilongTang @zeusongit @reddyashish 

### FYIs

@jesusalvino @RobertGlobant20

### CODE REVIEW

File: src/Notifications/NotificationCenterController.cs [hunk 1]
```diff
@@ -46,9 +56,45 @@ internal NotificationCenterController(DynamoView view, DynamoLogger dynLogger)
...
+            request.AutomaticDecompression = DecompressionMethods.GZip | DecompressionMethods.Deflate;
+
+            using (HttpWebResponse response = (HttpWebResponse)request.GetResponse())
+            using (Stream stream = response.GetResponseStream())
+            using (StreamReader reader = new StreamReader(stream))
+            {
+                jsonStringFile = reader.ReadToEnd();
+                notificationsModel = JsonConvert.DeserializeObject<NotificationsModel>(jsonStringFile);
+
+                var notificationsNumber = notificationsModel.Notifications.Count();
```
Comment (filipeotero) [inline src/Notifications/NotificationCenterController.cs:91]: This is counting all the notifications, but we may add a comparison between the read notification and the new ones.

File: src/Notifications/NotificationCenterController.cs [hunk 2]
```diff
@@ -46,9 +56,45 @@ internal NotificationCenterController(DynamoView view, DynamoLogger dynLogger)
...
             notificationUIPopup.webView.EnsureCoreWebView2Async();
             notificationUIPopup.webView.CoreWebView2InitializationCompleted += WebView_CoreWebView2InitializationCompleted;
             logger = dynLogger;
+
+            RequestNotifications();
+        }
+
+        private void WebView_NavigationCompleted(object sender, Microsoft.Web.WebView2.Core.CoreWebView2NavigationCompletedEventArgs e)
+        {
+            AddNotifications(notificationsModel.Notifications);
```
Comment (filipeotero) [inline src/Notifications/NotificationCenterController.cs:69]: We make sure navigation is completed before calling the js function

File: src/DynamoCoreWpf/ViewModels/Core/ShortcutToolbarViewModel.cs [hunk 3]
```diff
@@ -0,0 +1,14 @@
+﻿using Dynamo.ViewModels;
+
+namespace Dynamo.Wpf.ViewModels.Core
+{
+    public class ShortcutToolbarViewModel : ViewModelBase
```
Comment (QilongTang) [inline src/DynamoCoreWpf/ViewModels/Core/ShortcutToolbarViewModel.cs:5]: Other than the property, maybe the class can be internnal, please also add summary for these

File: src/Notifications/NotificationCenterController.cs (hunk 2 above)
Comment (QilongTang) [reply src/Notifications/NotificationCenterController.cs:69]: Is this because otherwise the js function may not exist?

Comment (filipeotero) [reply src/Notifications/NotificationCenterController.cs:69]: Yes, the navigation completed subscription makes sure the page content is loaded.

**Review by QilongTang — COMMENTED** [review]
These looks fine to me, what do you think @zeusongit @reddyashish ?

**Review by reddyashish — APPROVED** [review]

**Review by zeusongit — APPROVED** [review]

**Comment by QilongTang** [comment]
@filipeotero Would you cherry-pick this PR to RC branch?

**Pull request merged**